Automation/Python/Pytest tips & tricks.md

```diff
@@ -0,0 +1,91 @@
+## What is pytest?
+
+Pytest is a testing framework for Python.
+For more info, check the [documentation](https://pytest.org).
+
+
+### Conftest
+
+`conftest.py` is a file in which you can set up your test configuration. 
+The file should primarily consist of fixtures which are automatically detected and executed.
+
+You can have multiple `conftest` files in your project. 
+For example, you can have one `conftest` in the project root where you want to set up logic for the entire project. This one might be useful for the overall configuration, like setting up driver(s).
+
+And then you can add additional `conftest` file into your tests directory.
+In `tests/conftest.py` you could have fixtures that are used only on tests. Note that all tests will use those fixtures.
+
+You can even add another `conftest` in other test sub-directories.
+E.g. directory `test_a` could have its own conftest `tests/test_a/conftest.py`.
+
+The structure might look something like this:
+
+```python
+ROOT
+conftest.py
+└── tests
+    └── conftest.py
+    └── test_a
+        └── conftest.py
+        └── test_file_a.py
+    └── test_b
+        └── test_file_b.py
+
+```
+
+
+### Autouse
+
+Let's say there is some code that you want to run before or after every test but you do not want to call it every time. `autouse` option comes in handy. By setting the `autouse` parameter in a fixture to `True`, it is called automatically.
+
+
+```python
+@pytest.fixture(scope="function", autouse=True)
+"""Called automatically before every test."""
+    # Some code
+
+@pytest.fixture(scope="class", autouse=True)
+"""Called automatically after every class."""
```

Comment:
I updated the second example to make it a bit more easier to understand.
It now just yields and then does something.

It is actually called at the start, but when it sees `yield driver` it will just wait until driver is done doing its thing. Then execute the code.

Reply:
Thanks for clarifying it. The update that you made makes sense 🚀